date_range/tests/common.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+def setup_test_model(env, model_clses):
+    for model_cls in model_clses:
+        model_cls._build_model(env.registry, env.cr)
+
+    env.registry.setup_models(env.cr)
+    env.registry.init_models(
+        env.cr, [model_cls._name for model_cls in model_clses],
+        dict(env.context, update_custom_fields=True)
+    )
+
+
+def teardown_test_model(env, model_clses):
+    for model_cls in model_clses:
+        del env.registry.models[model_cls._name]
+        inherits = ["base"] + list(model_cls._inherits) + (
+            model_cls._inherit if isinstance(model_cls._inherit, list)
+            else [model_cls._inherit])
+        for inherited_cls in inherits:
+            if model_cls._name in env.registry.models[
+                    inherited_cls]._inherit_children:
+                env.registry.models[
+                    inherited_cls]._inherit_children.remove(
+                        model_cls._name)
```

Comment:
why not just iterate through all models then?

Reply:
Isn't this the more educated solution?

Reply:
as it works, don't change it, no